src/libs/memoize/types.ts

```diff
@@ -0,0 +1,34 @@
+type CacheMode = 'array' | 'map';
```

Comment:
Could you maybe add a comment somewhere about when you should use which mode?

Reply:
Sure thing! I have added comments, let me know if they are descriptive enough 🙏